### PR TITLE
Add a notification when temp-banned users reconnect

### PR DESCRIPTION
**Pull request by ShiftyRail**
image

Now we can !

bans

### CODE REVIEW

**Comment by DamianX** [comment]
Nice. Is there something similar for warnings as well? If not, there should be.
Also, since connecting is unreliable and messages have the risk of being sent but not received, it could be better to show this is as an `alert()` that has to be explicitly acknowledged by clicking OK or something like that.

File: code/modules/migrations/SS13/015-unbanned-var.dm [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/datum/migration/mysql/ss13/_014
```
Comment (ShiftyRail) [inline code/modules/migrations/SS13/015-unbanned-var.dm]: Wrong id...

**Comment by tuypo1** [comment]
I guarantee you we are going to be seeing admins ignoring this and doing it the old way anyway

**Comment by ShiftyRail** [comment]
@DamianX I think warnings can only be given if the client is connected (in my opinion, it makes no sense to warn someone who's not here), and they're using SQLite instead of MySQL. I don't feel like battling with it right now.

I can see the reasoning behind the `alert()` proc, but I fear it would be too much. I will wait for more opinions 

@tuypo1 for now, it's automatically enabled on all bans.

**Comment by tuypo1** [comment]
Well that's good to hear

**Comment by Kammerjunk** [comment]
sweetness
on a side note unrelated to this PR specifically, is grabbing query result cells by index the best option we have? not sure how DM does with JSON or if DM lets you get and set properties by string accessor, but I once made a query service which grabbed column names and converted the result to a JSON array. worked pretty well and wasn't a huge time investment

**Comment by ShiftyRail** [comment]
The original way of handling SQL in Byond was through the native procs, returning rows as normal lists. It was abysmally slow, so TG developed tools to offload that part of the code to an external tool, which they made return a JSON ordered by index that is transformed into a list Byond-side (Byond can only communicate with external dlls through raw strings.)

Grabbing columns names byond-side to make `query.NextRow()` return an assoc list instead of using magic indexes certainly is feasible. Better yet would be to change the external tool we use to make it return the columns names as well as the rows, but that requires going through the motions of making a PR on the TG repo (or to fork it), updating our dll, etc.

**Comment by Kammerjunk** [comment]
fair
also I meant a JSON array of JSON objects, of course

**Comment by ShiftyRail** [comment]
Added it for warning, will put in on secret repo too....

**Comment by DamianX** [comment]
players2.sql needs to be updated as well

**Pull request merged**